src/api/key_store.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use alloc::vec::Vec;
+use crypto::ecdsa::SecKey;
+use persistent_store::StoreError;
+use rng256::Rng256;
+
+use crate::env::Env;
+
+/// Provides storage for secret keys.
+///
+/// Implementations may use the environment store: [`STORE_KEY`] is reserved for this usage.
+pub trait KeyStore {
+    /// Returns the AES key for key handles encryption.
+    fn kh_encryption(&mut self) -> Result<[u8; 32], Error>;
```

Comment:
Yeah, I initially had a `_key` suffix and moved to `kh_` prefix because `_key` is very vague, it's not clear it's for key handle wrapping.

I don't like `credential_store` because that's false. This doesn't store credentials only (and it actually doesn't store any credential since the derive function is a function, it doesn't store the keys, it stores a secret to derive the keys from a seed).

Here are some naming suggestions:
- `KeyStore` -> `SecureStore` or `SecretStore` (although right now it's only use to provide keys)
- `kh_xxx` -> `key_handle_xxx_key` it's very explicit but maybe better? (I think we can drop the `_key` if we don't rename `KeyStore` since it's redundant).
- `kh_xxx` -> `xxx_key` (dropping the suffix if still using `KeyStore`) but this assumes that we only encrypt and authenticate key handles. It's maybe true? Since the rest is about signing AFAIU
- `kh_xxx` -> `master_xxx` with the implicit convention that master means key handle, which AFAICT is currently true (maybe the best option so far IMO)

Reply:
Thinking about it, I don't find anything much better than the alternatives, so I don't mind keeping it as is. The `KeyStore` alternatives are not better, and I see your point with `CredentialStore`.
`key_handle_` instead of `kh_` is more coherent with our style of being very sparse with acronyms. Your choice.

Reply:
Yes, I also prefer `key_handle_` more. Wasn't a big fan of `kh_` but `key_handle_` seemed quite long. But since it's confusing, it's still better to use the full name.